xarray/backends/zarr.py

```diff
@@ -317,6 +323,8 @@ def encode_zarr_variable(var, needs_copy=True, name=None):
     var = coder.encode(var, name=name)
     var = coding.strings.ensure_fixed_length_bytes(var)
 
+    if original_chunks and not var.chunks and "chunks" not in var.encoding:
+        var.encoding["chunks"] = _squeeze_var_chunks(original_chunks, name=name)
```

Comment:
I worry that fixing the issue this way reveals that our internal interfaces are leaky. It seems like a bandaid for a deeper problem.

Why does `encode_cf_variable` work for some dask-based variables but not for certain datetimes? Why does `CFDatetimeCoder` behave this way?  Is it possible that the encoder is eagerly computing the dask array by mistake?